### PR TITLE
fix(chat): fix duplication in compare mode and prompt deletion/folder structure duplication after share accepting from shared with me (Issues #850, #815, #979, #986)

### PR DESCRIPTION
**Pull request by Alexander-Kezik**
**Description:**

- Fix duplication in compare mode 
- Fix prompt deletion from shared with me
- Fix prompt folder structure duplication after share accepting

Issues:

- https://github.com/epam/ai-dial-chat/issues/850
- https://github.com/epam/ai-dial-chat/issues/815
- https://github.com/epam/ai-dial-chat/issues/979

**Checklist:**

- [x] the pull request name complies with [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/)
- [x] the pull request name starts with `fix(<scope>):`, `feat(<scope>):`, `feature(<scope>):`, `chore(<scope>):`, `hotfix(<scope>):` or `e2e(<scope>):`. If contains breaking changes then the pull request name must start with `fix(<scope>)!:`, `feat(<scope>)!:`, `feature(<scope>)!:`, `chore(<scope>)!:`, `hotfix(<scope>)!:` or `e2e(<scope>)!:` where `<scope>` is name of affected project: `chat`, `chat-e2e`, `overlay`, `shared`, `sandbox-overlay`, etc.
- [x] the pull request name ends with `(Issue #<TICKET_ID>)` (comma-separated list of issues)
- [x] I confirm that do not share any confidential information like API keys or any other secrets and private URLs

### CODE REVIEW

**Comment by Alexander-Kezik** [comment]
/deploy-review
>Environment URL: https://chat-ai-dial-chat-pr-980.nightly-test.deltixhub.io
>E2E tests status: failed

**Comment by Alexander-Kezik** [comment]
/deploy-review
>Environment URL: https://chat-ai-dial-chat-pr-980.nightly-test.deltixhub.io
>E2E tests status: success

**Comment by Alexander-Kezik** [comment]
/deploy-review
>Environment URL: https://chat-ai-dial-chat-pr-980.nightly-test.deltixhub.io
>E2E tests status: failed

File: apps/chat/src/store/conversations/conversations.epics.ts [hunk 1]
```diff
@@ -540,8 +541,8 @@ const createNewConversationsSuccessEpic: AppEpic = (action$) =>
 
 const saveNewConversationEpic: AppEpic = (action$) =>
   action$.pipe(
-    filter((action) => ConversationsActions.saveNewConversation.match(action)),
-    switchMap(({ payload }) =>
+    filter(ConversationsActions.saveNewConversation.match),
+    mergeMap(({ payload }) =>
```
Comment (IlyaBondar) [inline apps/chat/src/store/conversations/conversations.epics.ts:545]: Why do we need `mergeMap` here?

**Review by IlyaBondar — APPROVED** [review]

**Pull request merged**